constantine/eth_verkle_ipa/common_utils.nim

```diff
@@ -91,33 +91,34 @@ func computeInnerProducts* [Fr] (res: var Fr, a,b : View[Fr])=
 #
 # ############################################################
 
-func foldScalars* [Fr] (res: var openArray[Fr], a,b : openArray[Fr], x: Fr)=
+func foldScalars*(res: var openArray[Fr[Banderwagon]], a,b : openArray[Fr[Banderwagon]], x: Fr[Banderwagon])=
   ## Computes res[i] = a[i] + b[i] * x
-  debug: doAssert a.len == b.len , "Lengths should be equal!"
+  doAssert a.len == b.len , "Lengths should be equal!"
 
   for i in 0 ..< a.len:
-    var bx {.noInit.}: Fr
-    bx.prod(x, b[i])
-    res[i].sum(bx, a[i])
+    var bx {.noInit.}: Fr[Banderwagon]
+    bx.prod(b[i], x)
+    res[i].sum(a[i], bx)
 
-func foldPoints* [EC_P] (res: var openArray[EC_P], a,b : var openArray[EC_P], x: Fr)=
+func foldPoints*(res: var openArray[EC_P], a,b : openArray[EC_P], x: Fr[Banderwagon])=
   ## Computes res[i] = a[i] + b[i] * x
-  debug: doAssert a.len == b.len , "Should have equal lengths!"
+  doAssert a.len == b.len , "Should have equal lengths!"
 
   for i in 0 ..< a.len:
     var bx {.noInit.}: EC_P
-
-    b[i].scalarMul(x.toBig())
     bx = b[i]
-    res[i].sum(bx, a[i])
+    var x_big {.noInit.}: matchingOrderBigInt(Banderwagon)
+    x_big = x.toBig()
```

Comment:
```suggestion
    x_big.fromField(x)
```
or keep as before.
Alternatively a `scalarMul` overload that accepts field elements